File: app/src/main/java/com/fireflylearning/tasksummary/persistence/PersistenceManager.java
package com.fireflylearning.tasksummary.persistence;


import com.fireflylearning.tasksummary.model.CustomLiveData;
import com.fireflylearning.tasksummary.persistence.entities.Task;

import java.util.List;

/**
 * Created by Roll on 11/8/17.
 */

public interface PersistenceManager {

    void insertTask(Task task);

    void insertListOfTasks(List<Task> tasks);

    void loadTasks(CustomLiveData<List<Task>> liveTasks);

}

File: app/src/main/java/com/fireflylearning/tasksummary/persistence/PersistenceManager.kt
package com.fireflylearning.tasksummary.persistence

import android.arch.lifecycle.LiveData
import android.arch.paging.PagedList
import com.fireflylearning.tasksummary.persistence.entities.Task

/**
 * Created by Roll on 5/10/17.
 * interface for the database helper
 */
interface PersistenceManager {

    fun insertTask(task: Task)

    fun insertListOfTasks(tasks: List<Task>)

    fun loadTasks() : LiveData<PagedList<Task>>

}